### PR TITLE
More tests + minor refactoring and cleanup

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/UrlMatcher.php [hunk 1]
```diff
@@ -73,18 +66,16 @@ final class UrlMatcher implements UrlMatcherInterface
...
         CacheInterface $cache = null,
-        array $config = null,
+        ?array $config = null,
         RouteCollector $fastRouteCollector = null,
         callable $dispatcherFactory = null
     ) {
-        if (null === $fastRouteCollector) {
-            $fastRouteCollector = $this->createRouteCollector();
-        }
+        $fastRouteCollector = $fastRouteCollector ?? $this->createRouteCollector();
```
Comment (rustamwin) [inline src/UrlMatcher.php:78]:
```suggestion
        $this->fastRouteCollector = $fastRouteCollector ?? $this->createRouteCollector();
```

Comment (vjik) [reply src/UrlMatcher.php]: Done

File: src/UrlMatcher.php [hunk 2]
```diff
@@ -23,12 +22,12 @@
 final class UrlMatcher implements UrlMatcherInterface
 {
     /**
-     * @const string Configuration key used to set the cache file path
+     * Configuration key used to set the cache file path
```
Comment (samdark) [inline src/UrlMatcher.php:25]:
```suggestion
     * Configuration key used to set the cache file path.
```

File: src/UrlMatcher.php [hunk 3]
```diff
@@ -23,12 +22,12 @@
...
      */
     public const CONFIG_CACHE_KEY = 'cache_key';
 
     /**
-     * @const string Configuration key used to set the cache file path
+     * Configuration key used to set the cache file path
```
Comment (samdark) [inline src/UrlMatcher.php:30]:
```suggestion
     * Configuration key used to set the cache file path.
```

**Review by samdark — APPROVED** [review]

**Pull request merged**